### ISSUE TITLE
Speed up CMake builds

### ISSUE THREAD

**Issue by bfowle** [clustergarage/argusd#27]
CMake builds are slow, especially on DockerHub automated builds (45min+).
Only build necessary ExternalProject components, don't link where not needed, don't run tests when not needed, etc.

**Comment by bfowle** [comment]
- [ ] `c-ares` and `zlib` are required to build the gRPC we use; they provide two methods to BYO or compile during the gRPC cmake build: `-DXXX_PROVIDER=[package|module]` unfortunately this doesn't seem to work currently without having to build as an external project first then BYO with `package`
- [ ] gRPC seems to need both `zlib` static/shared versions to compile (?)
- [ ] `openssl` is currently being built and included in gRPC as well, instead of the `boringssl` due to a boringssl bug
- [ ] turning `-DgRPC_BUILD_CODEGEN` to OFF does not work due to a bug

tl;dr-- gRPC cmake is a nightmare

**Comment by bfowle** [comment]
TODO:
- [ ] using `-DCMAKE_C_COMPILER=gcc` without full path complains on gRPC package (only local build, not docker)
- [ ] `CMP0048` policy warnings (`-Wno-dev` suppression?)
- [ ] test with `-DCMAKE_CXX_COMPILER` as `clang` and `llvm`
- [ ] benchmark comparison
  - [ ] different generators
  - [ ] different compilers
